doc: cleanup doc on need of Developer Account to obtain macOS SDK

jarolrod opened the pull request:
The explicit statement that an Apple Developer Account is required in order to obtain the SDK is buried within the `Deterministic macOS DMG Notes` section. It should be the first thing mentioned under the `SDK Extraction` section.
The reason to do this is to set expectations of what is required before starting any steps or clicking on links.

This fixes the issue by doing just that; moving this information to the `SDK Extraction` section. Now that the information is moved, this also deletes unnecessary SDK related notes from the `Deterministic macOS DMG Notes` section. It is not necessary to explain under what sub-directory 'most' of the important files are inside of the `Xcode.app`.

Note that this also fixes a missed Xcode version link bump by deleting it :)

jarolrod committed:
doc: cleanup doc on need of Developer Account to obtain macOS SDK

The explicit statement that an Apple Developer Account is required in
order to obtain the SDK is buried within the "Deterministic macOS DMG
Notes" section. It should be the first thing mentioned under the "SDK
Extraction" section. The reason to do this is to set expectations of
what is required before starting any steps or clicking on links.

This fixes the issue by doing just that; moving this information to the
"SDK Extraction" section. Now that the information is moved, this also
deletes unnecessary SDK related notes from the "Deterministic macOS DMG
Notes" section. It is not necessary to explain under what sub-directory
'most' of the important files are inside of the 'Xcode.app'.

RandyMcMillan commented:
NIT:

commit: https://github.com/RandyMcMillan/bitcoin/commit/118f821037f4ff4ef078f88a8cbbd4f88c70786d

update the term "look for" to "search for" with the actual search link.

This is the result of clicking on the updated link:

![Screen Shot 2022-02-02 at 5 24 51 PM](https://user-images.githubusercontent.com/152159/152247872-3d8c4591-c05a-4c60-82bd-30ef0a2123bc.png)


commit: https://github.com/RandyMcMillan/bitcoin/commit/118f821037f4ff4ef078f88a8cbbd4f88c70786d as patch:

```
diff --git a/contrib/macdeploy/README.md b/contrib/macdeploy/README.md
index ef8c1f430c..31288284b7 100644
--- a/contrib/macdeploy/README.md
+++ b/contrib/macdeploy/README.md
@@ -19,7 +19,7 @@ Our current macOS SDK
 extracted from
 [Xcode_12.2.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.2/Xcode_12.2.xip).
 Alternatively, after logging in to your account go to 'Downloads', then 'More'
-and look for [`Xcode_12.2`](https://download.developer.apple.com/Developer_Tools/Xcode_12.2/Xcode_12.2.xip).
+and search for [`Xcode 12.2`](https://developer.apple.com/download/all/?q=Xcode%2012.2).
 An Apple ID and cookies enabled for the hostname are needed to download this.
 The `sha256sum` of the archive should be `28d352f8c14a43d9b8a082ac6338dc173cb153f964c6e8fb6ba389e5be528bd0`.

```

Sjors commented on contrib/macdeploy/README.md:
```diff
@@ -12,6 +12,8 @@ When complete, it will have produced `Bitcoin-Core.dmg`.
 
 ### Step 1: Obtaining `Xcode.app`
 
+An Apple Developer Account is required to proceed.
```
Maybe: "A free Apple Developer Account" (to clarify that you don't need to sign up for their paid app developer program)

Sjors reviewed (commented):
Concept ACK

@RandyMcMillan's suggestion to replace the link with a search URL makes sense. Apple doesn't believe in statelessness, so the direct link doesn't work until you perform the right rituals on their site.

brunoerg reviewed (commented):
Concept ACK

I agree with @Sjors in mentioning that Apple Developer Account is free.

fanquake commented:
Concept ACK - can you incorporate the suggestions here

jarolrod commented:
updated from [b2dcf36](https://github.com/bitcoin/bitcoin/commit/b2dcf36663f3bc7b25e1588dd65597f41c2f3374) -> [ddcac22](https://github.com/bitcoin/bitcoin/commit/ddcac22f092de6f3a2a67acb3288a990fd13cb01) ([pr24241.01](https://github.com/jarolrod/bitcoin/tree/pr24241.01) -> [pr24241.02](https://github.com/jarolrod/bitcoin/tree/pr24241.02), [diff](https://github.com/jarolrod/bitcoin/compare/pr24241.01..pr24241.02))

Thanks everyone for the reviews! Incorporated review feedback, here's a convenient diff of the changes:
```diff
diff --git a/contrib/macdeploy/README.md b/contrib/macdeploy/README.md
index ef8c1f430..ce69079e2 100644
--- a/contrib/macdeploy/README.md
+++ b/contrib/macdeploy/README.md
@@ -12,14 +12,14 @@ When complete, it will have produced `Bitcoin-Core.dmg`.
 
 ### Step 1: Obtaining `Xcode.app`
 
-An Apple Developer Account is required to proceed.
+A free Apple Developer Account is required to proceed.
 
 Our current macOS SDK
 (`Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers.tar.gz`) can be
 extracted from
 [Xcode_12.2.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.2/Xcode_12.2.xip).
 Alternatively, after logging in to your account go to 'Downloads', then 'More'
-and look for [`Xcode_12.2`](https://download.developer.apple.com/Developer_Tools/Xcode_12.2/Xcode_12.2.xip).
+and search for [`Xcode_12.2`](https://developer.apple.com/download/all/?q=Xcode%2012.2).
 An Apple ID and cookies enabled for the hostname are needed to download this.
 The `sha256sum` of the archive should be `28d352f8c14a43d9b8a082ac6338dc173cb153f964c6e8fb6ba389e5be528bd0`.
```

fanquake reviewed (approved):
ACK ddcac22f092de6f3a2a67acb3288a990fd13cb01
